File: src/main/java/dev/psygamer/wireframe/util/collection/FreezableVector.java
package dev.psygamer.wireframe.util.collection;

import java.util.Collection;
import java.util.Vector;

public class FreezableVector <E> extends FreezableList<E> {
	
	public FreezableVector() {
		super(new Vector<>());
	}
	
	public FreezableVector(final int capacity) {
		super(new Vector<>(capacity));
	}
	
	public FreezableVector(final int capacity, final int capacityIncrement) {
		super(new Vector<>(capacity, capacityIncrement));
	}
	
	public FreezableVector(final Collection<? extends E> c) {
		super(new Vector<>(c));
	}
	
	@Override
	public FreezableList<E> copy() {
		final FreezableList<E> list = new FreezableVector<>(this);
		
		if (isFrozen())
			list.freeze();
		
		return list;
	}
}

File: src/main/kotlin/dev/psygamer/wireframe/util/collection/FreezableVector.kt
package dev.psygamer.wireframe.util.collection

import java.util.*

class FreezableVector<E> : FreezableList<E> {
	
	constructor() : super(Vector<E>())
	constructor(capacity: Int) : super(Vector<E>(capacity))
	constructor(capacity: Int, capacityIncrement: Int) : super(Vector<E>(capacity, capacityIncrement))
	constructor(elements: Collection<E>) : super(Vector<E>(elements))
}